File: app/src/main/java/org/standardserve/driftflasche/login/textValidation.java
package org.standardserve.driftflasche.login;

import android.text.TextUtils;
import android.util.Log;

import java.util.Objects;

// Class to validate text input
public class textValidation {

    private textValidation(){}

    // Method to validate email
    public static boolean emailValidation(String email) {
        return !TextUtils.isEmpty(email) && android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches();
    }

    // Method to validate password
    public static boolean passwordValidation(String password){
        return password.length() >= 6;
    }

    // Method to validate username
    public static boolean repeatPasswordValidation(String password, String repeatPassword){
        return Objects.equals(password, repeatPassword);
    }

    // Method to validate username
    public static boolean truenameValidation(String truename){
        return truename.length() >= 2 && truename.length() <= 20;
    }

    // Method to validate comment length
    public static boolean limitCommentLength(String comment){
        return comment.length() <= 20;
    }

    // Method to validate bottle text length
    public static boolean limitBottleTextLength(String bottleText){
        return bottleText.length() <= 100;
    }
}

File: app/src/main/java/org/standardserve/driftflasche/login/textValidation.kt
package org.standardserve.googletestunit.login

import android.text.TextUtils

object textValidation {
    fun emailValidation(email: String) = !TextUtils.isEmpty(email) && android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches()
    fun passwordValidation(password: String) = password.length >= 6
    fun repeatPasswordValidation(password: String, repeatPassword: String) = password == repeatPassword
    fun truenameValidation(truename: String) = truename.length in 2..20
}